tests/unit/utils/test_imageheader.py

```diff
@@ -0,0 +1,63 @@
+# vim: ft=python fileencoding=utf-8 sw=4 et sts=4
+
+# This file is part of vimiv.
+# Copyright 2023-2023 Christian Karl (karlch) <karlch at protonmail dot com>
+# License: GNU GPL v3, see the "LICENSE" and "AUTHORS" files for details.
+
+"""Tests for vimiv.utils.imageheader."""
+
+from PyQt5.QtGui import QPixmap, QImageReader
+
+import pytest
+
+
+from vimiv.utils import imageheader
+
+# Image types which QPixmap is capable of creating, and not what is supported by vimiv!
+SUPPORTED_IMAGE_FORMATS = ["jpg", "png", "pbm", "pgm", "ppm", "bmp"]
+
+
+@pytest.fixture()
+def mockimageheader(mocker):
+    """Fixture to mock imageheader._registry and QImageReader supportedImageFormats."""
+    mocker.patch.object(
+        QImageReader, "supportedImageFormats", return_value=SUPPORTED_IMAGE_FORMATS
+    )
+    yield mocker.patch("vimiv.utils.imageheader._registry", [])
+
+
+@pytest.fixture()
+def tmpfile(tmp_path):
+    path = tmp_path / "anything"
+    path.touch()
+    yield str(path)
+
+
+def create_image(filename: str, *, size=(300, 300)):
+    QPixmap(*size).save(filename)
+
+
+@pytest.mark.parametrize("imagetype", SUPPORTED_IMAGE_FORMATS)
+def test_test(qtbot, tmp_path, imagetype):
```

Comment:
Mind calling this `test_detect` instead?

Reply:
That's not by purpose. I started writing that function when I did not know what that function is gonna be about. And once I did know, I forgot about my dummy naming 🙈